### PR TITLE
Ensure that the various `Max*Number` and `Min*Number` APIs optimize correctly

### LINKED ISSUE #98068: `INumber.MinNumber` and `INumber.MaxNumber` return `NaN` in `Release` configuration
### Description

When running a program in `Release` configuration, `INumber.MinNumber` and `INumber.MaxNumber` return `NaN` if one of the numbers is not `NaN`. In `Debug` configuration these methods work as expected.

### Reproduction Steps

1. Create a new console project, specify `.NET 8` as the target platform.
2. Add the following code:

``` csharp
int numbers = 0;
int NaN = 0;

for (int i = 0; i < 100000; i++)
{
    float minLength = 0;
    float maxLength = 0;

    if (!float.IsNaN(float.MaxNumber(maxLength / minLength, 1)))
    {
        numbers++;
    }
    else
    {
        NaN++;
    }
}

Console.WriteLine($"Numbers: {numbers}");
Console.WriteLine($"NaN: {NaN}");
```
3. Run the program.

### Expected behavior

The output of the program should be as follows:

```
Numbers: 100000
NaN: 0
```

### Actual behavior

Here is the result of the program in `Release` configuration:

```
Numbers: 9999
NaN: 90001
```

### Regression?

The bug could not be reproduced in `.NET 7`. The program works as expected in both configurations.

### Known Workarounds

You can use `INumberBase.MinMagnitudeNumber` and `INumberBase.MaxMagnitudeNumber` if you intend to work with only positive numbers.

### Configuration

.NET:
Microsoft .NET SDK 8.0.101 (x64) from Visual Studio (8.1.123.58017)
Microsoft .NET SDK 8.0.200-preview.23624.5 (x64) from Visual Studio (8.2.23.62405)

OS:
Windows 11 Pro 23H2 22631.3085
Windows 11 Pro 22H2 22621.3085

Architecture:
x64
x64

### Other information

The bug is not reproduced if you replace the first argument of `MaxNumber` with one of the following:

``` csharp
if (!float.IsNaN(float.MaxNumber(0f / 0f, 1)))
{
    numbers++;
}
else
{
    NaN++;
}
```

``` csharp
if (!float.IsNaN(float.MaxNumber(float.NaN, 1)))
{
    numbers++;
}
else
{
    NaN++;
}
```

If we run the code in [SharpLab](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQ6CWVgAIoQC2ARgKZgDOuAvLgAwDcWOuAcgIas31MZoBmAezC4AFNjyZujXFIA8uJHWXKGsgNTqAlOgDe6XIdz8ANoPZ5C2ADKkoAc2AALabyPGzF3IXYAPWw7OrugGRpj8YgCEpubAAHQAkhQcrKIxFnEAsn6sRGRgoj7+do4uAPTeNiXOADSKWg2hhvpo7u4EJOQUmm5GAL5NuKQmFKSDLW1GKT2DA2hz6HBIAJyiACQARLmdlCC4uh35FH0bWrxLq5spe7opJ2dAA===) and [Compiler Explorer](https://godbolt.org/z/x7Te1YaaW), we get 100,000 `NaN`. On my machines I get 9,999 numbers and 90,001 `NaN`.

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #98068

### CODE REVIEW

File: src/tests/JIT/Regression/JitBlue/Runtime_98068/Runtime_98068.cs [hunk 1]
```diff
@@ -0,0 +1,754 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+public static class Runtime_98068
+{
+    [Fact]
+    [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.AggressiveOptimization)]
```
Comment (EgorBo) [inline src/tests/JIT/Regression/JitBlue/Runtime_98068/Runtime_98068.cs:12]: I think most pipelines disable TC anyway so there is no need in AggressiveOptimization. At the same time it'd be nice to make sure that Tier0 (hence, the non-intrinsic impl) passes these as well

**Comment by EgorBo** [comment]
I assume this might slightly regresses codegen since `vmaxss(reg, cns)` is better than `vmaxss(cns, reg)` since the former can "contain" cns load. But it's a correctness fix so 🤷

**Comment by tannergooding** [comment]
Right.

**Review by EgorBo — APPROVED** [review]
LGTM assuming CI passes

**Pull request merged**